.github/workflows/screenshots.yml

```diff
@@ -0,0 +1,79 @@
+name: Screenshots
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main    
+
+jobs:
+  screenshots:
+    
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - uses: actions/setup-node@v1
+      with:
+        node-version: 12
+    - uses: actions/cache@v2
+      with:
+        path: ~/.npm
+        key: node-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: node-    
+    - name: install
+      run: npm ci
+    - name: run
+      run: |
+        npm run watch & 
+        sleep 60s
+      env: 
+        APIS_IMG_SALT: ${{ secrets.APIS_IMG_SALT }}  
+        CAPI_KEY: ${{ secrets.CAPI_KEY }}
+    - name: Take screenshot of apps-rendering
+      uses: "lannonbr/puppeteer-screenshot-action@1.3.0"
+      with:
+        url: http://localhost:8080/
+    - name: Upload screenshot
+      run: aws s3 cp screenshots/screenshot*.png s3://gu-mobile-pr-images/${{ github.sha }}/screenshot.png --acl public-read
+      env:
+        AWS_ACCESS_KEY_ID: ${{ secrets.CI_IMAGE_UPLOADER_ACCESS_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.CI_IMAGE_UPLOADER_SECRET_ACCESS_KEY }}
+    - name: Prepare image urls
```

Comment:
This is all getting a bit ugly - we could write our own [Javascript action](https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/about-actions#types-of-actions) (and store it within this repo) if this bothers people too much.

Reply:
This sounds like a nice solution longer term, if we'd like to port this functionality to other projects.